### PR TITLE
Add solution to cf 1316 e

### PR DESCRIPTION
**Pull request by TruaShamu**

### CODE REVIEW

**Comment by TruaShamu** [comment]
Could someone help me check what I did wrong please?

It says that the internal solution hasn't been linked to all of it's problems, but I'm pretty sure I've edited the json already, so I'm not sure where I screwed up...

**Comment by dongliu0426** [comment]
apparently the problem is also in https://usaco.guide/adv/min-cost-flow?lang=cpp

**Comment by TruaShamu** [comment]
thanks for fixing it for me :)

File: solutions/CF Team Building.mdx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+source: CF
+title: Team Building
+author: Sofia Yang
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/74493)
+
+## Solution
+
+**Time Complexity:** $\mathcal{O}(N\cdot p\cdot 2^N)$
```
Comment (dolphingarlic) [inline solutions/CF Team Building.mdx:12]: Do you mean 2^p?

File: content/6_Advanced/MinCostFlow.problems.json [hunk 2]
```diff
@@ -49,8 +49,7 @@
       "isStarred": false,
       "tags": [],
       "solutionMetadata": {
-        "kind": "link",
-        "url": "https://codeforces.com/blog/entry/74493?#comment-586020"
+        "kind": "internal"
```
Comment (dolphingarlic) [inline content/6_Advanced/MinCostFlow.problems.json:52]: I think you should leave this as it was before because your solution doesn't use min cost flow

**Review by dolphingarlic — CHANGES_REQUESTED** [review]
I haven't checked the code because I don't know Java

File: content/6_Advanced/MinCostFlow.problems.json (hunk 2 above)
Comment (dongliu0426) [reply content/6_Advanced/MinCostFlow.problems.json:52]: oops i added this. removing this breaks the deployment though.

Comment (dolphingarlic) [reply content/6_Advanced/MinCostFlow.problems.json:52]: Huh that's weird

Comment (dolphingarlic) [reply content/6_Advanced/MinCostFlow.problems.json:52]: @thecodingwizard @jeffreymeng any idea why this happens?

Comment (maggie-j-liu) [reply content/6_Advanced/MinCostFlow.problems.json:52]: This is intended behavior:
> We assume that if there is an internal solution, we should use it; therefore, the build will throw an error if there is an internal solution but the `solutionMetadata`'s `kind` isn't set to `'internal'`.

https://usaco.guide/general/working-mdx/#problem-lists

File: solutions/CF Team Building.mdx (hunk 1 above)
Comment (TruaShamu) [reply solutions/CF Team Building.mdx]: Yes, sorry, it's fixed now.

File: content/6_Advanced/MinCostFlow.problems.json (hunk 2 above)
Comment (thecodingwizard) [reply content/6_Advanced/MinCostFlow.problems.json:52]: It's intended. Maybe just add this link to the internal solution?

Comment (TruaShamu) [reply content/6_Advanced/MinCostFlow.problems.json:52]: So how should I fix it, or should I just leave it as-is?

Comment (dolphingarlic) [reply content/6_Advanced/MinCostFlow.problems.json:52]: Add a "solution 2" thing under the official solution and link the cf comment for now

File: solutions/CF Team Building.mdx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+author: Sofia Yang
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/74493)
+
+## Solution
+
+**Time Complexity:** $\mathcal{O}(N\cdot p\cdot 2^p)$
+
+## Implementation
```
Comment (dongliu0426) [inline solutions/CF Team Building.mdx:14]:
```suggestion
### Implementation
```

**Review by dongliu0426 — CHANGES_REQUESTED** [review]

File: solutions/CF Team Building.mdx [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+author: Sofia Yang
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/74493)
+
+## Solution
+
+**Time Complexity:** $\mathcal{O}(N\cdot p\cdot 2^p)$
+
+### Implementation
```
Comment (maggie-j-liu) [inline solutions/CF Team Building.mdx:14]:
```suggestion
### Implementation

**Time Complexity:** $\mathcal{O}(N\cdot p\cdot 2^p)$
```

**Review by dongliu0426 — APPROVED** [review]

File: solutions/CF Team Building.mdx [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+					}
+				}
+
+				//Add the player to the team.
+				for (int j = 0; j < P; j++) {
+					if ((m & (1 << j)) > 0 ) {
+						if ((dp[i - 1][m ^ (1 << j)]) != -1) {
+
+							dp[i][m] = Math.max(dp[i][m],
+									dp[i - 1][m ^ (1 << j)] + skill[ind][j]);
```
Comment (maggie-j-liu) [inline solutions/CF Team Building.mdx:89]: this line is way too long, but there's really no way to shorten it so it should be fine...

File: solutions/CF Team Building.mdx [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+						}
+					}
+				}
+			}
+		}
+
+		//Print the answer.
+		System.out.println(dp[N][(1<< P)- 1]);
```
Comment (maggie-j-liu) [inline solutions/CF Team Building.mdx:97]:
```suggestion
		System.out.println(dp[N][(1 << P) - 1]);
```

File: solutions/CF Team Building.mdx [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+<JavaSection>
+
+```java
+import java.util.*;
+import java.io.*;
+public class Main {
+
+	public static void main(String[] args) throws IOException {
+		BufferedReader r =
+		new BufferedReader(new InputStreamReader(System.in));
```
Comment (maggie-j-liu) [inline solutions/CF Team Building.mdx:27]:
```suggestion
			new BufferedReader(new InputStreamReader(System.in));
```
maybe this looks better?

File: content/6_Advanced/MinCostFlow.problems.json [hunk 8]
```diff
@@ -49,8 +49,7 @@
       "isStarred": false,
       "tags": [],
```
Comment (maggie-j-liu) [inline content/6_Advanced/MinCostFlow.problems.json:50]: maybe add "Bitmasks" and "MinCostFlow" ?

File: content/6_Advanced/MinCostFlow.problems.json [hunk 9]
```diff
@@ -47,10 +47,9 @@
       "source": "CF",
       "difficulty": "Hard",
       "isStarred": false,
-      "tags": [],
+      "tags": ["Bitmasks", "MCF"],
```
Comment (maggie-j-liu) [inline content/6_Advanced/MinCostFlow.problems.json:50]:
```suggestion
      "tags": ["Bitmasks", "MinCostFlow"],
```
i think this is the name of the existing tag

**Review by maggie-j-liu — APPROVED** [review]

**Pull request merged**